provider/server/machine_create_test.go

```diff
@@ -24,13 +28,86 @@ var _ = Describe("CreateMachine", func() {
 				Spec: &iri.MachineSpec{
 					Power: iri.Power_POWER_ON,
 					Image: &iri.ImageSpec{
-						Image: "example.org/foo:latest",
+						Image: imageURL,
 					},
-					Class: machineClassx3xlarge,
+					Class:        machineClassx3xlarge,
+					IgnitionData: ignitionData,
+					// TODO: will be done when convertMachineToIRIMachine() supports Volumes and NetworkInterfaces
+					Volumes:           nil,
+					NetworkInterfaces: nil,
 				},
 			},
 		})
 		Expect(err).NotTo(HaveOccurred())
-		Expect(res).NotTo(BeNil())
+
+		By("ensuring the correct creation response")
+		Expect(createResp).Should(SatisfyAll(
+			HaveField("Machine.Metadata.Id", Not(BeEmpty())),
+			HaveField("Machine.Spec.Power", iri.Power_POWER_ON),
+			HaveField("Machine.Spec.Image.Image", Equal(imageURL)),
+			HaveField("Machine.Spec.Class", machineClassx3xlarge),
+			HaveField("Machine.Spec.IgnitionData", Equal(ignitionData)),
+			HaveField("Machine.Spec.Volumes", BeNil()),
+			HaveField("Machine.Spec.NetworkInterfaces", BeNil()),
+			HaveField("Machine.Status.ObservedGeneration", BeZero()),
+			HaveField("Machine.Status.State", Equal(iri.MachineState_MACHINE_PENDING)),
+			HaveField("Machine.Status.ImageRef", BeEmpty()),
+			HaveField("Machine.Status.Volumes", BeNil()),
+			HaveField("Machine.Status.NetworkInterfaces", BeNil()),
+		))
+
+		DeferCleanup(func(ctx SpecContext) {
```

Comment:
Because the cleanup will be needed in multiple test files in this suite, I recommend moving common cleanup steps into a separate function for better readability and maintainability, which can be used everywhere else, to avoid rewriting code.

Reply:
@afritzler Could you please let us know your view on this.

Reply:
it is more nice to have for me but it isn't important. We can do improvement always later and i prefer working tests.